logger.d.ts

```diff
@@ -0,0 +1,49 @@
+// Type definitions for js-logger.js
+// Project: http://github.com/jonnyreeves/js-logger
+
+interface ILogLevel {
+    value: number;
+    name: string;
+}
+
+interface ILogContext {
+    name: string;
+    level: ILogLevel;
+    filterLevel: ILogLevel;
+}
+
+
+interface ILogger {
+  setLevel(newLevel: ILogLevel): void;
+  enabledFor(level: ILogLevel): boolean;
+  debug(...args: any[]): void;
+  info(...args: any[]): void;
+  log(...args: any[]): void;
+  warn(...args: any[]): void;
+  error(...args: any[]): void;
+  time(label: string): void;
+  timeEnd(label: string): void;
+}
+
+
+interface ContextualLogger extends ILogger {
+  invoke(level: ILogLevel, ...args: any[]): void;
+  context: ILogContext;
+}
+
+
+interface Logger extends ILogger {
+  get(name: string): ContextualLogger;
+  useDefaults(level?: ILogLevel): void;
+  setHandler(handler: (args: any[], 
```

Comment:
prefer no line break here - function signatures in typescript can get long, but line breaking them makes them harder to read.